modin/utils.py

```diff
@@ -258,30 +297,31 @@ def _replace_doc(
     target_doc = target_obj.__doc__ or ""
     overwrite = overwrite or not target_doc
     doc = source_doc if overwrite else target_doc
-    apilink = [apilink] if not isinstance(apilink, list) and apilink else apilink
+    apilink_l: Optional[List[str]]
+    apilink_l = [apilink] if not isinstance(apilink, list) and apilink else apilink  # type: ignore
```

Comment:
The logic for the isinstance check is a little weird:
if (apilink is not a list) and (apilink is truth-y): apilink_l = apilink
else: apilink = [apilink]

The code as written means that if `apilink` is `None`, it would fall into the second case, meaning`apilink_l = None`. This is checked by an if statement later before the variable is read.

The `type: ignore` is there because without it, the "else" branch can be taken if `apilink` is the empty string, making the type of the expression `Union[List[str], str, None]`. I'm not sure if there's a cleaner way to rewrite this logic to prevent that.

Reply:
What if you made it `Optional[Union[List[str], str]]` ?

Reply:
The code later on expects a list when iterating over `apilink_l` (though now that I think about it, since strings also act like lists it might typecheck fine). I'll fix this by moving the declaration of `apilink_l` further down so it can't be None.